pkg/uploader/provider/provider.go

```diff
@@ -49,23 +54,34 @@ type Provider interface {
 		volumePath string,
 		updater uploader.ProgressUpdater) error
 	// Close which will close related repository
-	Close(ctx context.Context)
+	Close(ctx context.Context) error
 }
 
 // NewUploaderProvider initialize provider with specific uploaderType
 func NewUploaderProvider(
 	ctx context.Context,
+	client client.Client,
 	uploaderType string,
 	repoIdentifier string,
 	bsl *velerov1api.BackupStorageLocation,
-	backupReo *velerov1api.BackupRepository,
+	backupRepo *velerov1api.BackupRepository,
 	credGetter *credentials.CredentialGetter,
 	repoKeySelector *v1.SecretKeySelector,
 	log logrus.FieldLogger,
 ) (Provider, error) {
+	if credGetter.FromFile == nil {
+		return nil, errors.New("uninitialized FileStore credentail is not supported")
+	}
 	if uploaderType == uploader.KopiaType {
-		return NewResticUploaderProvider(repoIdentifier, bsl, credGetter, repoKeySelector, log)
+		if err := provider.NewUnifiedRepoProvider(*credGetter, log).ConnectToRepo(ctx, provider.RepoParam{BackupLocation: bsl, BackupRepo: backupRepo}); err != nil {
```

Comment:
Why do we need to create repo provider here?

Reply:
For every `NewUploaderProvider`, which is running in the daemonset, It's a start of backup&restore  for every `PVB` or `PVR`, we need to use the latest configuration in BSL to check the repository connection at first when provider initializes.